cmd/nfs_threads.py

```diff
@@ -132,35 +143,44 @@ def print_thruput(value):
         print('{0:>8}KB'.format(int(value / 1024)), end='')
 
 
-def print_line():
+def print_line(interval):
+    lines = 0
     pids = nfsd_processes()
 
-    prevSwitches = context_switches(pids)
-    prevPackets, prevEnqueued, prevWoken, prevTimedout = pool_stats()
-    prevRB, prevWB, prevRO, prevWO, prevMeta = nfs_stats()
+    prevCpuTime, e1 = cpu_time(pids)
+    prevPackets, prevEnqueued, prevWoken, e2 = pool_stats()
+    prevRB, prevWB, prevRO, prevWO, prevMeta, e3 = nfs_stats()
+    if e1 or e2 or e3:
+        return
```

Comment:
That's an interesting way to pass errors ;).
I'm curious, could you just propagate/re-throw exception to the caller of `print_line` and let it call `server_stopped()` and continue?

Reply:
There is initialization that is needed at the start of `print_line` so it can't use a `continue` here.  It is effectively two nested while loops where the outer loop is outside of `print_line` .